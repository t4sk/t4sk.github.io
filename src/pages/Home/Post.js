import React from "react"
import PropTypes from "prop-types"
import { Image } from "semantic-ui-react"
import styles from "./Post.module.css"

export function Post(props) {
  const { youTubeUrl, thumbnailUrl, title, createdAt } = props

  return (
    <a className={styles.component} href={youTubeUrl} target="__blank">
      <div className={styles.image}>
        <Image src={thumbnailUrl} fluid />
      </div>
      <div className={styles.content}>
        <div className={styles.title}>{title}</div>
        <div className={styles.date}>{createdAt}</div>
      </div>
    </a>
  )
}

Post.propTypes = {
  youTubeUrl: PropTypes.string.isRequired,
  thumbnailUrl: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  createdAt: PropTypes.string.isRequired,
}

export default Post
