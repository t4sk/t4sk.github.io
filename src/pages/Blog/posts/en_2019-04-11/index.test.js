import React from "react"
import { shallow } from "enzyme"
import Post from "./index"

test("en", () => {
  const component = shallow(<Post />)

  expect(component).toMatchSnapshot()
})
