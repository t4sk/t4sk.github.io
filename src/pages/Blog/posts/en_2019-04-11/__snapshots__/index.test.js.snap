// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`en 1`] = `
<PostLayout
  lang="en"
  title="Solidity by Example - Part 0 - Setting Up MetaMask"
>
  <_default />
</PostLayout>
`;
